services/slam/builtin/builtin_test.go

```diff
@@ -43,7 +43,7 @@ import (
 )
 
 const (
-	timePadding      = 5
+	timePadding      = 20
```

Comment:
We run the data process for (validDataRateMS+timePadding)*5ms and expect 5 images to be written. Changing timePadding to 20ms should make the total time 1100ms, which should not cause 6 images to be written.

Reply:
Valid change. Do we want to instead make it dependent on the validDataRateMS? i.e.

`time.Sleep(time.Millisecond * time.Duration((n+0.5)*validDataRateMS))`

Reply:
I don't have a strong preference either way. Either way, you need to know n when setting timePadding, to make sure that n*timePadding < validDataRateMS, so that you don't generate an extra image.

Reply:
Also, I think we would want to factor the padding fraction out into a constant, so we would still end up with the same number of constants.